projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -149,6 +163,9 @@ abstract public static function get_pricing_for_ui();
 	public static function get_status() {
 		if ( static::is_active() ) {
 			$status = 'active';
+			if ( ! static::has_required_plan() ) {
+				$status = 'needs_purchase';
+			}
```

Comment:
something we've been talking about. I thought.
I though we could check whether the product requires a plan at top of this method:

```suggestion
	public static function get_status() {
		if ( ! static::has_required_plan() ) {
			$status = 'needs_purchase';
		} elseif ( static::is_active() ) {
			$status = 'active';
```

It's ok to me, just wanted to put in code the way that we'll have to take to deal with the product activation flow.

Reply:
`has_required_plan` will return true by default. If the product doesn't need a plan, it will return true.

But I agree that could be useful to have another property that specifically says if the product is free or not